Attack refactor

Tableu opened the pull request.

fokai7 committed:
click ui sfxs

fokai7 committed:
Merge branch 'master' of https://github.com/Tableu/Bismarck

fokai7 committed:
meta

Tableu committed:
Removing scripts

Tableu committed:
Deleting prefabs and assets

maddok7 committed:
yes!

Added Credits

maddok7 committed:
Added Title

Am000zing committed:
UI stuff + created new folder in sprites

maddok7 committed:
Added and edited!

maddok7 committed:
Additions and editions!

maddok7 committed:
Loss piece

Tableu committed:
Init

Tableu committed:
Revising AttackCommands, Replacing IDamageable with DamageableComponent abstract class

Tableu committed:
Merge branch 'experimental' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

 Conflicts:
	Assets/Scripts/Attacks/PointLaserOrb.cs
	Assets/Scripts/Ships/Components/DamageableComponentInfo.cs
	Assets/Scripts/Ships/Components/ShipInfo.cs
	Assets/Scripts/Ships/DataManagement/ShipData.cs
	Assets/Scripts/Ships/IDamageable.cs

Tableu committed:
Merge changes

Tableu committed:
Weapon Firing Test

Tableu committed:
Setting up MapView scene

Tableu committed:
Merge branch 'master' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

Tableu committed:
Merge meta files

Am000zing committed:
Create UIbutton.png

Am000zing committed:
damaged buttons

Tableu committed:
Removing outdated scripts, adding script to spawn infowindows and some prefabs

Tableu committed:
Reorganizing infowindow files. Fixing enemy fleet data

Tableu committed:
Adding positioning for ship visuals

Tableu committed:
Merge branch 'master' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

Tableu committed:
Adding weapon firing, setting up MapView init

Tableu committed:
Organizing folders, adding comments

Tableu committed:
Further cleanup

Tableu committed:
New art

Am000zing committed:
ahah sfskjdhfj

Am000zing committed:
doot doot turrets

Tableu committed:
Adding events for component disable/enable, ship death.

Tableu committed:
planet

Tableu committed:
planet metas

Am000zing committed:
updato

Tableu committed:
Adding InfoWindow animations

Tableu committed:
Merge branch 'master' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

Tableu committed:
New Art

Tableu committed:
Fixing WeaponButtons background

Tableu committed:
Setting up Abilities

Tableu committed:
Adding HealthBars to InfoWindow and MapIcons

Tableu committed:
Fixed DamageableComponent health error

Tableu committed:
Setting up Ability buttons

Tableu committed:
Implementing Ability bar

Tableu committed:
Adding ButtonData and abilityButtons prefab

Tableu committed:
Forgot scene

Tableu committed:
Linking abilities to ShipInfo

Am000zing committed:
button updates and turret button UI

Am000zing committed:
Create dragon turret B button.png

KerbalJeb commented:
Could you add a brief description of the changes & a quick expiation of any demo scenes & what the should do.

Tableu commented:
- Refactoring Attack System to start/end in InfoWindow and travel through the MapView to the target. Deleting various prefabs and scripts that rely on colliders.
- Implementing DamageableComponent: monobehavior scripts which are attached to the ship by ShipInfo and can be targeted by weapons.
- Implementing Subsystem, Attack and Ability button bars: Horizontal layout groups of buttons that are spawned based off a list of objects. Each button performs the same function on distinct objects (setting weapon targets, firing a weapon, etc)
- Set up InfoWindow spawning in MapView: Clicking on an enemy ship will spawn an infowindow in the top right.
- Added necessary initialization for enemies/player to ShipInfo and FleetManagers/Load enemies
- Demo Scene is MapView: everything should be interactable on play, click on enemy ships to open infowindows and click on subsystem buttons to target them. If a target is selected when an attack button is pressed it should fire a projectile

Tableu committed:
Adding initializer to AttackAnimation, splitting speed into map speed and infowindow speed. Adding new Attack

Tableu committed:
Updating data

KerbalJeb commented:
The MapView scene seems to be missing prefabs & does not appear to work.
![image](https://user-images.githubusercontent.com/14899798/157370875-2da6db95-b240-4b9a-a415-c547d234886d.png)

Tableu committed:
Setting up Miss animation

Tableu committed:
Fixing MapView Scene

Tableu committed:
Merge branch 'master' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

Tableu committed:
Subsystem Button icons

KerbalJeb reviewed (commented):
**Main thoughts**
- I'm not quite sure I understand the separation of *AttackCommands*, *Weapons* and *AttackData*. I feel like weapons should be split into two parts: the attack ability & the damageable component. We also need to finish designing the ship customization systems to figure out exactly how it'll work, but I think it's safe to say that you'll want to have something like a *weapon slot* that you can put different weapons into.

- I think that attacks should be abilities (ie. a targetable action) so that there is once consistent way of handling cool-downs

- Should move everything related to abilities out of ship info and into an ability manager class

- Should make a monobehavour to aid in targeting ships/subsystems that also determines if a ship is hostile or not (so you can activate or de-active abilities appropriately). 


**Bugs**
- You can shoot your self if you select hull/systems on your own ship
- Ship info windows can be opened multiple times for each ship

Tableu committed:
Splitting weapons into attacks and damageable component.
Created Cooldown ability class for abilities and attacks.
Merged Weapon and ability buttons/managers
Moved abilities out of ship stats
Renaming some things

Tableu committed:
Setting up audio, fixing MapView scene

Tableu committed:
Asset files

Tableu committed:
Merging Weapons into Abilities

Tableu committed:
Addressing smaller comments

Tableu committed:
Disabling ability buttons depending on target.

Tableu committed:
Reworking MapView UI out of logic classes and into its own logic that read off of data independently.

Tableu committed:
Adding RangeIndicators

Tableu committed:
Adding tags, merging weapon/ability cases in AbilityBar

Tableu committed:
Moving targeting logic to TargetingHelper

Tableu committed:
Adding comments, polish, reorganizing

Tableu committed:
Fixing AttackIcon stall bug, deleting weapon button prefabs

Tableu committed:
Improving fire/miss/hit animations to hit specific modules, adding visuals reference to damageable components, deleting unnecessary files

maddok7 committed:
Added, changed, organized; travelbattlestandby piece

Tableu committed:
Merge branch 'master' of https://github.com/Tableu/Roguelike-Autochess into attack-refactor

 Conflicts:
	Assets/Music/Boss Battle 1 - Vanya_vs._The World.wav.meta

Tableu committed:
Adding sounds

Tableu committed:
Adding music
